pkg/operator/controller.go

```diff
@@ -785,341 +572,90 @@ func (c *Controller) getMachineConfigNamesForTuned() (map[string]bool, error) {
-			object, ok := o.(metav1.Object)
-			if !ok {
-				tombstone, ok := o.(cache.DeletedFinalStateUnknown)
-				if !ok {
-					klog.Errorf("error decoding object, invalid type")
-					return
-				}
-				object, ok = tombstone.Obj.(metav1.Object)
-				if !ok {
-					klog.Errorf("error decoding object tombstone, invalid type")
-					return
-				}
-				klog.V(4).Infof("recovered deleted object %s from tombstone", object.GetName())
-			}
-			if clusterOperator, ok := o.(*configapiv1.ClusterOperator); ok {
-				if clusterOperator.GetName() != tunedv1.TunedClusterOperatorResourceName {
-					return
-				}
-			}
-			klog.V(2).Infof("add event to workqueue due to %s (delete)", util.ObjectInfo(object))
-			c.workqueue.Add(wqKey{kind: workqueueKey.kind, namespace: object.GetNamespace(), name: object.GetName()})
-		},
-	}
-}
-
-// enableNodeInformer enables/disables event handling for Nodes.
-func (c *Controller) enableNodeInformer(enable bool) error {
-	if (enable && c.node.informerEnabled) || (!enable && !c.node.informerEnabled) {
-		return nil
-	}
-
-	if enable {
-		var (
-			informerFactory kubeinformers.SharedInformerFactory
-			informer        corev1informers.NodeInformer
-		)
-		c.node.stopCh = make(chan struct{})
-		informerFactory = kubeinformers.NewSharedInformerFactoryWithOptions(c.clients.Kube, ntoconfig.ResyncPeriod(), kubeinformers.WithNamespace(corev1.NamespaceAll))
-
-		informer = informerFactory.Core().V1().Nodes()
-		c.listers.Nodes = informer.Lister()
-		informer.Informer().AddEventHandler(c.informerEventHandler(wqKey{kind: wqKindNode}))
-
-		informerFactory.Start(c.node.stopCh)
-	} else {
-		defer close(c.node.stopCh)
-		c.node.stopCh <- struct{}{}
-		c.pc.nodeLabelsDelete()
-	}
-
-	c.node.informerEnabled = enable
-	return nil
-}
-
-// enablePodInformer enables/disables event handling for Pods.
-func (c *Controller) enablePodInformer(enable bool) error {
-	if (enable && c.pod.informerEnabled) || (!enable && !c.pod.informerEnabled) {
-		return nil
-	}
-
-	if enable {
-		var (
-			informerFactory kubeinformers.SharedInformerFactory
-			informer        corev1informers.PodInformer
-		)
-		c.pod.stopCh = make(chan struct{})
-		informerFactory = kubeinformers.NewSharedInformerFactoryWithOptions(c.clients.Kube, ntoconfig.ResyncPeriod(), kubeinformers.WithNamespace(corev1.NamespaceAll))
-
-		informer = informerFactory.Core().V1().Pods()
-		c.listers.Pods = informer.Lister()
-		informer.Informer().AddEventHandler(c.informerEventHandler(wqKey{kind: wqKindPod}))
-
-		informerFactory.Start(c.pod.stopCh)
-	} else {
-		defer close(c.pod.stopCh)
-		c.pod.stopCh <- struct{}{}
-		c.pc.podLabelsDelete()
-	}
-
-	c.pod.informerEnabled = enable
-	metrics.PodLabelsUsed(enable)
-	return nil
-}
-
-func (c *Controller) removeResources() error {
+func (r *TunedReconciler) removeResources(ctx context.Context) error {
 	var lastErr error
+
 	dsMf := ntomf.TunedDaemonSet()
-	ctx := context.TODO()
+	ds := &appsv1.DaemonSet{}
+	key := types.NamespacedName{
```

Comment:
The deletion removes the DS but the operator will re-create it.

Reply:
good point.  this is handling for the ManagementState : Removed  but the CVO reconciler will just recreate it.  Probably this logic should be applied and synced on the CVO controller .

Reply:
@cynepco3hahue   the CVO reconciler logic is now changed to first act on managed state.
It will first inspect the ManagementState. In case the operator is under Unmanaged or Removed: update the status and take no further actions.
see https://github.com/openshift/cluster-node-tuning-operator/pull/302/commits/81de5a6f332cc0eed5e106187dfebf9510a499dd